src/Microsoft.AspNetCore.Mvc.Razor.Host/Internal/ViewComponentTagHelperChunkVisitor.cs

```diff
@@ -0,0 +1,206 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Chunks;
+using Microsoft.AspNetCore.Razor.CodeGenerators;
+using Microsoft.AspNetCore.Razor.CodeGenerators.Visitors;
+using Microsoft.AspNetCore.Razor.Compilation.TagHelpers;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Host.Internal
+{
+    public class ViewComponentTagHelperChunkVisitor : CodeVisitor<CSharpCodeWriter>
+    {
+        private readonly GeneratedViewComponentTagHelperContext _context;
+        private readonly HashSet<string> _writtenViewComponents;
+
+        private const string _viewComponentHelperVariable = "_viewComponentHelper";
+        private const string _viewContextVariable = "ViewContext";
```

Comment:
const strings don't have underscores under them, standard for us to have these capitalized.